server/market/bookrouter.go

```diff
@@ -0,0 +1,412 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package market
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+
+	"github.com/decred/dcrdex/server/comms"
+	"github.com/decred/dcrdex/server/comms/msgjson"
+	"github.com/decred/dcrdex/server/market/types"
+	"github.com/decred/dcrdex/server/order"
+)
+
+// A bookUpdateAction classifies updates into how they affect the book or epoch
+// queue.
+type bookUpdateAction uint8
+
+const (
+	// invalidAction is the zero value action and should be considered programmer
+	// error if received.
+	invalidAction bookUpdateAction = iota
+	// epochAction means an order is being added to the epoch queue and will
+	// result in a msgjson.EpochOrderNote being sent to subscribers.
+	epochAction
+	// bookAction means an order is being added to the order book, and will result
+	// in a msgjson.BookOrderNote being sent to subscribers.
+	bookAction
+	// unbookAction means an order is being removed from the order book and will
+	// result in a msgjson.UnbookOrderNote being sent to subscribers.
+	unbookAction
+)
+
+// bookUpdateSignal combines a bookUpdateAction with the order on which the
+// action applies.
+type bookUpdateSignal struct {
+	action bookUpdateAction
```

Comment:
`bookUpdateSignal` and `bookUpdateAction` need to be exported so `Market` can make and send the signals

Reply:
This is the `market` package.